src/components/modals/PrintPreminedModal.tsx

```diff
@@ -0,0 +1,159 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { Form, Input, Modal, Switch } from 'antd'
+import { useForm } from 'antd/lib/form/Form'
+import { FormItems } from 'components/shared/formItems'
+import { ProjectContext } from 'contexts/projectContext'
+import { UserContext } from 'contexts/userContext'
+import { constants, Contract, utils } from 'ethers'
+import { CurrencyOption } from 'models/currency-option'
+import { useContext, useMemo, useState } from 'react'
+import { parseWad } from 'utils/formatNumber'
+
+export default function PrintPreminedModal({
+  visible,
+  onCancel,
+}: {
+  visible: boolean | undefined
+  onCancel: VoidFunction
+}) {
+  const { contracts, transactor } = useContext(UserContext)
+  const { tokenSymbol, tokenAddress, projectId, terminal } =
+    useContext(ProjectContext)
+  const [form] = useForm<{
+    beneficary: string
+    preferUnstaked: boolean
+    memo: string
+  }>()
+
+  const [currency, setCurrency] = useState<CurrencyOption>(0)
+  const [value, setValue] = useState<string>('0')
+  const [loading, setLoading] = useState<boolean>()
+
+  async function mint() {
+    if (!contracts || !projectId || !transactor || !terminal?.version) return
+
+    setLoading(true)
+
+    await form.validateFields()
+
+    let terminalContract: Contract
+    let functionName: string
+    let args: any[]
+
+    switch (terminal.version) {
+      case '1':
```

Comment:
**suggestion (non-blocking)**: worth having shared constants for these?

Reply:
maybe, these are typed strings so an error will get thrown if you're using anything that isn't a version. i don't think it's as important here as it was with currency, where 0/1 does nothing to indicate ETH/USD

Reply:
agreed. Given it's typed, should be fine 👍